etf/encode.go

```diff
@@ -403,6 +403,20 @@ func Encode(term Term, b *lib.Buffer,
 			}
 			lenString := len(t)
 
+			if lenString > 4294967295 {
```

Comment:
```suggestion
			if lenString > math.MaxUint32 {
```

Reply:
left along the idea to keep the 4GB string I would encode etf.String into the binary and would introduce etf.Charlist to replace current logic of stringAsCharlist

Reply:
See latest commit.